### PR TITLE
fix VPC peering connection import

### LINKED ISSUE #751: Unable to terraform import `aiven_aws_vpc_peering_connection` resources
When I try to run

```
terraform import aiven_aws_vpc_peering_connection.my_connections[\"vpc-012345\"] my_project/my_vpc/my_aws_account_id/vpc-012345
```
I see the following error:

```
Error: invalid identifier my_project/my_vpc/my_aws_account_id/vpc-012345, expected <project_name>/<vpc_id>/<aws_account_id>
```

Looking at the documentation this matches the suggested [import code](https://registry.terraform.io/providers/aiven/aiven/latest/docs/resources/aws_vpc_peering_connection#import). However, trying
```
terraform import aiven_aws_vpc_peering_connection.my_connections my_project/my_vpc/my_aws_account_id
```
gives the following error:

```
aiven_aws_vpc_peering_connection.my_connections: Importing from ID "my_project/my_vpc/my_aws_account_id"...
|
| Error: Plugin did not respond
| 
| The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ImportResourceState call. The plugin logs may contain more details.
|


Stack trace from the terraform-provider-aiven_v3.0.0 plugin:

panic: runtime error: index out of range [3] with length 3

goroutine 71 [running]:
github.com/aiven/terraform-provider-aiven/internal/service/vpc.parsePeeringVPCId(0xc0000484c0, 0x39, 0xc0005114d8, 0x44e30c, 0x30, 0xcde660, 0x1, 0xc0005e4930, 0xc000511530, 0x4f79ff, ...)
        github.com/aiven/terraform-provider-aiven/internal/service/vpc/resource_vpc_peering_connection.go:234 +0x1d6
github.com/aiven/terraform-provider-aiven/internal/service/vpc.resourceAWSVPCPeeringConnectionRead(0xf769e0, 0xc0005e4780, 0xc000a25400, 0xdc2ba0, 0xc0001906e0, 0x2, 0xc0005e4930, 0x3)
        github.com/aiven/terraform-provider-aiven/internal/service/vpc/resource_aws_vpc_peering_connection.go:160 +0x97
github.com/aiven/terraform-provider-aiven/internal/service/vpc.resourceAWSVPCPeeringConnectionImport(0xf769e0, 0xc0005e4780, 0xc000a25400, 0xdc2ba0, 0xc0001906e0, 0x12, 0x0, 0x0, 0x0, 0x20)
        github.com/aiven/terraform-provider-aiven/internal/service/vpc/resource_aws_vpc_peering_connection.go:228 +0x259
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Provider).ImportState(0xc000114360, 0xf769e0, 0xc0005e4780, 0xc000511768, 0xc0000484c0, 0x39, 0x9, 0xc000000001, 0xe216ec, 0x12, ...)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/provider.go:385 +0x544
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ImportResourceState(0xc00098c360, 0xf76938, 0xc0005e4780, 0xc000c24fe0, 0xe216ec, 0x12, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/grpc_provider.go:1088 +0x11a
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ImportResourceState(0xc000176000, 0xf769e0, 0xc0005e4060, 0xc0005e22d0, 0x0, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.9.0/tfprotov5/tf5server/server.go:841 +0x2ed
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ImportResourceState_Handler(0xddfb20, 0xc000176000, 0xf769e0, 0xc0005e4060, 0xc000115200, 0x0, 0xf769e0, 0xc0005e4060, 0xc00073e900, 0x5d)
        github.com/hashicorp/terraform-plugin-go@v0.9.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:403 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00013e540, 0xf7e598, 0xc0003c81a0, 0xc0009d9680, 0xc0003cb1d0, 0x1435358, 0x0, 0x0, 0x0)
        google.golang.org/grpc@v1.46.0/server.go:1283 +0x544
google.golang.org/grpc.(*Server).handleStream(0xc00013e540, 0xf7e598, 0xc0003c81a0, 0xc0009d9680, 0x0)
        google.golang.org/grpc@v1.46.0/server.go:1620 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0002ba000, 0xc00013e540, 0xf7e598, 0xc0003c81a0, 0xc0009d9680)
        google.golang.org/grpc@v1.46.0/server.go:922 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.46.0/server.go:920 +0x1fd

Error: The terraform-provider-aiven_v3.0.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

I've tried a number of different combinations of the command, but all result in an error similar to this.

The import command that I used for the older version of this resource `aiven_vpc_peering_connection` used the syntax that I tried first, which is why I tried that first. Looking at the import command for similar commands for [Azure](https://registry.terraform.io/providers/aiven/aiven/latest/docs/resources/azure_vpc_peering_connection#import) and [GCP](https://registry.terraform.io/providers/aiven/aiven/latest/docs/resources/gcp_vpc_peering_connection#import) also seem to show that the peering connection ID is required.

It looks like this is missing from the `aiven_aws_vpc_peering_connection` resource.

### PR DESCRIPTION
**Pull request by Serpentiel**
fixes #751

### CODE REVIEW

File: examples/resources/aiven_vpc_peering_connection/import.sh [hunk 1]
```diff
@@ -1 +1,2 @@
+terraform import aiven_vpc_peering_connection.mypeeringconnection project/vpc_id/peer_cloud_account/peer_vpc
```
Comment (dangquangdon) [inline examples/resources/aiven_vpc_peering_connection/import.sh:1]: Nit: Perhaps adding some comment about what use case of each import statement would be nice

File: internal/service/vpc/resource_aws_vpc_peering_connection.go [hunk 2]
```diff
@@ -221,8 +221,8 @@ func resourceAWSVPCPeeringConnectionDelete(ctx context.Context, d *schema.Resour
 }
 
 func resourceAWSVPCPeeringConnectionImport(ctx context.Context, d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
-	if len(strings.Split(d.Id(), "/")) != 3 {
-		return nil, fmt.Errorf("invalid identifier %v, expected <project_name>/<vpc_id>/<aws_account_id>", d.Id())
+	if len(strings.Split(d.Id(), "/")) < 4 {
+		return nil, fmt.Errorf("invalid identifier %v, expected <project_name>/<vpc_id>/<aws_account_id>/<aws_vpc_id>[/<aws_vpc_region>]", d.Id())
```
Comment (dangquangdon) [inline internal/service/vpc/resource_aws_vpc_peering_connection.go:225]: Should we have different format of error message for each `import` format?

**Review by dangquangdon — CHANGES_REQUESTED** [review]
Some small comments and perhaps we should include tests for these import statements

File: internal/service/vpc/resource_aws_vpc_peering_connection.go (hunk 2 above)
Comment (Serpentiel) [reply internal/service/vpc/resource_aws_vpc_peering_connection.go:225]: I used [ch. 12 of IEEE Std 1003.1](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap12.html) to specify an optional parameter in the error message.

Comment (dangquangdon) [reply internal/service/vpc/resource_aws_vpc_peering_connection.go:225]: would this always fail for this import format? As we would have the `len(strings.Split(d.Id(), "/")) == 3`
`terraform import aiven_aws_vpc_peering_connection.foo project_name/vpc_id/aws_account_id/aws_vpc_id`

**Comment by Serpentiel** [comment]
> Some small comments and perhaps we should include tests for these import statements

there's currently no way to test VPC since it's a bit tricky and after a small discussion with @ivan-savciuc and @mhoffm-aiven we decided to add VPC tests later and for now we'll be only testing whether the identifier validation works or not

**Comment by ivan-savciuc** [comment]
I will test it manually today to make sure that everything is OK

**Comment by dangquangdon** [comment]
> > Some small comments and perhaps we should include tests for these import statements
> 
> there's currently no way to test VPC since it's a bit tricky and after a small discussion with @ivan-savciuc and @mhoffm-aiven we decided to add VPC tests later and for now we'll be only testing whether the identifier validation works or not
:+1:  Cool, so feel free to dismiss my change request. And btw let's rebase the branch to `origin/master` so that it's up to date

**Review by ivan-savciuc — CHANGES_REQUESTED** [review]

**Review by ivan-savciuc — COMMENTED** [review]
Please squash all the commits into one and re-base on the latest master

**Comment by Serpentiel** [comment]
done

File: internal/service/vpc/resource_vpc_peering_connection.go [hunk 3]
```diff
@@ -37,7 +37,7 @@ var aivenVPCPeeringConnectionSchema = map[string]*schema.Schema{
 	},
 	"peer_region": {
 		ForceNew: true,
-		Optional: true,
+		Required: true,
```
Comment (ivan-savciuc) [inline internal/service/vpc/resource_vpc_peering_connection.go:40]: Sorry, perhaps I wasn't clear enough about making `peer_region` required. This resource is deprecated and coming from v2. It combines all types of VPC peering connections in one, like AWS, Azure and GCP. Therefore we mustn't make `peer_region` as required.

File: internal/service/vpc/resource_vpc_peering_connection.go [hunk 4]
```diff
@@ -361,8 +361,16 @@ func resourceVPCPeeringConnectionDelete(ctx context.Context, d *schema.ResourceD
 }
 
 func resourceVPCPeeringConnectionImport(ctx context.Context, d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
-	if len(strings.Split(d.Id(), "/")) != 4 {
-		return nil, fmt.Errorf("invalid identifier %v, expected <project_name>/<vpc_id>/<peer_cloud_account>/<peer_vpc>", d.Id())
+	if len(strings.Split(d.Id(), "/")) != 5 {
```
Comment (ivan-savciuc) [inline internal/service/vpc/resource_vpc_peering_connection.go:364]: Lets keep here `if len(strings.Split(d.Id(), "/")) != 4 {`, see comment from above.

File: internal/service/vpc/resource_gcp_vpc_peering_connection.go [hunk 5]
```diff
@@ -206,9 +206,17 @@ func resourceGCPVPCPeeringConnectionImport(ctx context.Context, d *schema.Resour
 		return nil, fmt.Errorf("invalid identifier %v, expected <project_name>/<vpc_id>/<gcp_project_id>/<peer_vpc>", d.Id())
 	}
 
+	client := m.(*aiven.Client)
+
+	projectName, vpcID, peerCloudAccount, peerVPC, peerRegion := parsePeeringVPCId(d.Id())
```
Comment (ivan-savciuc) [inline internal/service/vpc/resource_gcp_vpc_peering_connection.go:211]: GCP has a different ID format, without `peerRegion`

**Review by ivan-savciuc — CHANGES_REQUESTED** [review]
AWS resource looks good, please check comments related to other resource

File: internal/service/vpc/resource_gcp_vpc_peering_connection.go (hunk 5 above)
Comment (Serpentiel) [reply internal/service/vpc/resource_gcp_vpc_peering_connection.go:211]: it returns `*string` and will be `nil` there, which won't be passed to the API

**Pull request merged**